src/stdlib_stringlist.f90

```diff
@@ -0,0 +1,726 @@
+                                                                insert_before_stringarray_int
+        ! procedure         :: get_string_int         => get_string_int_impl
+        procedure         :: get_string_idx         => get_string_idx_wrap
+        generic, public   :: get                    => get_string_idx
+                                                        ! get_string_int
+
+    end type stringlist_type
+
+    !> Version: experimental
+    !> 
+    !> Concatenates stringlist with the input entity
+    !> Returns a new stringlist
+    !> [Specifications](../page/specs/stdlib_stringlist.html#append-operator)
+    interface operator(//)
+        module procedure append_char
+        module procedure append_string
+        module procedure prepend_char
+        module procedure prepend_string
+        module procedure append_stringlist
+        module procedure append_carray
+        module procedure append_sarray
+        module procedure prepend_carray
+        module procedure prepend_sarray
+    end interface
+
+    !> Version: experimental
+    !> 
+    !> Compares stringlist for equality with the input entity
+    !> Returns a logical
+    !> [Specifications](../page/specs/stdlib_stringlist.html#equality-operator)
+    interface operator(==)
+        module procedure eq_stringlist
+        module procedure eq_stringlist_carray
+        module procedure eq_stringlist_sarray
+        module procedure eq_carray_stringlist
+        module procedure eq_sarray_stringlist
+    end interface
+
+    !> Version: experimental
+    !> 
+    !> Compares stringlist for inequality with the input entity
+    !> Returns a logical
+    !> [Specifications](../page/specs/stdlib_stringlist.html#inequality-operator)
+    interface operator(/=)
+        module procedure ineq_stringlist
+        module procedure ineq_stringlist_carray
+        module procedure ineq_stringlist_sarray
+        module procedure ineq_carray_stringlist
+        module procedure ineq_sarray_stringlist
+    end interface
+
+contains
+
+    !> Returns an instance of type 'stringlist_index_type' representing forward index 'idx'
+    pure function forward_index(idx)
+        integer, intent(in) :: idx
+        type(stringlist_index_type) :: forward_index
+
+        forward_index = stringlist_index_type( .true., idx )
+
+    end function forward_index
+
+    !> Returns an instance of type 'stringlist_index_type' representing backward index 'idx'
+    pure function backward_index(idx)
+        integer, intent(in) :: idx
+        type(stringlist_index_type) :: backward_index
+
+        backward_index = stringlist_index_type( .false., idx )
+
+    end function backward_index
+
+    !> Returns a deep copy of the stringlist 'original'
+    pure function create_copy( original )
+        class(stringlist_type), intent(in)  :: original
+        type(stringlist_type)               :: create_copy
+
+        create_copy = original
+
+    end function create_copy
```

Comment:
Is there a reason to have an explicit `copy` procedure if intrinsic assignment already makes a copy? Do you envision changes at some point in the future that would cause assignment to no longer make a full, deep copy?

Reply:
I added it to make it more explicit.
I preferred it over normal assignment operator, because it helped me mention the motive. Someone who is reading the code might miss this motive of the developer otherwise. You won't find me using an assignment operator to do deep copying anywhere in the PR.
Otherwise, there is no specific reason for doing so and even it makes the procedure to take more time.

If you wish I can remove it?

Reply:
`copy` is not part of the public API. But used internally to create deep copies when appending lists to lists. I think the motivation here is to avoid destroying the list in case of aliasing for an operation like:
```f90
list = list // list
```

Reply:
The semantics of the language and your API's (i.e. `intent(in)`) already guarantee that you're doing a deep copy that things like `list = list // list` will work fine. You can see it in that the implementation of `create_copy` just uses intrinsic assignment, and that intrinsic assignment works for `new_list = list%copy()`. In principle that latter one is doing a copy twice (although in practice the compiler is probably smart enough to only do it once). Unless you switch the internal representation to using pointers - which would cause intrinsic assignment to not make a deep copy - I don't see any need to have an explicit copy procedure.

Reply:
Ok, I will remove the copy function and add some comments there instead to mention my motive.

After reading Sebastian's comment of `list // list`, I realised that inserting a list in the same list will cause errors.
```
list%insert_at( fidx(x), list )
```